licenses/MIT.lre

```diff
@@ -2,26 +2,95 @@
 MIT License
 https://spdx.org/licenses/MIT.json
 https://opensource.org/licenses/MIT
+https://fedoraproject.org/wiki/Licensing:MIT
 **//
 
 (( MIT License))??
 
 //**Copyright**//
 
-Permission is hereby granted, free of charge, to any person obtaining a copy of
-this software and associated documentation files (the "Software"), to deal in
-the Software without restriction, including without limitation the rights to
-use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
-the Software, and to permit persons to whom the Software is furnished to do so,
-subject to the following conditions:
-
-The above copyright notice and this permission notice
-((including the next paragraph))??
-shall be included in all copies or substantial portions of the Software.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
-FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL __5__ BE LIABLE
-FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
-TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR
-THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+Permission is hereby granted,
+((free of charge))??
+to any person obtaining a copy of
+__7__ //** (allow parenthetical descriptions) **//
+((and associated documentation files))??
+((the || this))
+((Software || Materials))
+to deal in the
+((Software || Materials))
+((under the copyrights))??
+((without restriction))??
+including
+((without limitation))??
+the rights
+((to))??
+use, copy, modify, merge, publish, distribute,
+((sub-license || sublicense))??
+and/or
+((sell))??
+((modified))??
+copies of the
+((Software || Materials))
+and to permit persons to whom the
+((Software is || Materials are))
+furnished to do so,
+subject to
+((the following || all))
+conditions:
+
+__1__
+((
+	The above
+	((copyright || authorship))
+	notice
+	(( and this permission notice
+		((including the next paragraph))??
+	|| as well as this permission notice
+	|| this permission notice, and the below disclaimer
+	|| and every other copyright notice found in this software,
+		and all the attributions in every file, and this permission notice
+	|| and this permission notice (or reference to this permission notice) ))
+||
+	This permission notice
+))
+((must || shall))
+be included in all
+((copies || copy))
+or
+((substantial || any))??
+portions of the
+((Software || Materials))
+
+((DISCLAIMER))??
+
+__1__
+
+{{template "mit-disclaimer"}}
+
+{{define "mit-disclaimer"}}
+THE
+((SOFTWARE IS || MATERIALS ARE))
+PROVIDED "AS IS",
+WITHOUT WARRANTY OF ANY KIND,
+EXPRESS OR IMPLIED,
+INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE
+AND NONINFRINGEMENT.
+IN NO EVENT
+((SHALL||WILL))
+__5__ BE LIABLE
+FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+WHETHER IN AN ACTION OF CONTRACT,
+((TORT || FART))
```

Comment:
This one is borderline but I don't mind accommodating the Onion.
I'd rather fail on the cases that are really not the MIT license.
This one's fine.

Reply:
I bet this is just for fartscroll, but let it be an Easter egg.

Reply:
By the way, I wasn't sure whether this entire approach of writing these regular expressions was going to scale well enough at all. It seemed like maybe, but maybe not, especially as I kept trying more real-world licenses. I waited to send any of this code out until I was confident it was going to work as an approach. I have 51 more commits waiting to be sent out after these 8, all license additions, to cover nearly every SPDX license—well over 300—and matching against a corpus I built of of 47k license files containing 64k licenses.

This LRE for the MIT license handles 37k real-world instances of unique license files containing the MIT license text in some form. Clearly it required many fewer than 37k adjustments. So I think it does end up scaling and not being madness, based on that experiment.

Also, MIT and BSD turn out to be by far the fussiest licenses to recognize, because they have been in use for such a long time, and because they date from a time when licenses were short and company lawyers felt no qualms about making cosmetic changes to match their sense of legal style. Today, licenses are longer (so less attractive to edit) and most IP lawyers now agree to just leave the standard wordings alone. So the newer licenses have dramatically less variation, despite being also dramatically longer.